### PR TITLE
Story 1 

### PR DESCRIPTION
**Pull request by ellie-r**

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * connects to the database and extracts all items required and returns in an associative array
+ * @param $dbName, the name of the db to connect to
+ *
+ * @return array, the array of items in the collection
+ */
+function connect_to_db_and_extract_items($dbName)
```
Comment (mporam) [inline functions.php]: a function should only do one thing. if you need an "and" in the function name, its a good indicator that it should be two functions

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    return $query->fetchAll();
+}
+
+/**
+ * takes items from an array and returns the values within HTML
+ * @param array $dataFromQuery, array of data from database
+ *
+ * @return string, return string of html
+ */
+
```
Comment (mporam) [inline functions.php:37]: no line break between docblock and function

File: functions.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * connects to the database and extracts all items required and returns in an associative array
+ * @param $dbName, the name of the db to connect to
+ *
+ * @return array, the array of items in the collection
+ */
+function connect_to_db_and_extract_items($dbName)
+{
+    $db = new PDO('mysql:host=db;dbname=' . $dbName, 'root', 'password'); //initialise the db connection
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    $query = $db->prepare('SELECT `variety`, `tones`, `cost`, `nameOfBrand`, `country`, `wine`.`img_location` FROM `wine` JOIN `brands` ON `wine`.`brand_id` = `brands`.`id` JOIN `country` ON `wine`.`country_of_origin`=`country`.`id`;');
```
Comment (mporam) [inline functions.php]: phpstorm has a vertical line about 80 characters into your screen, this line represents how long a line of code can be before you should wrap it onto several lines. this is one of those times

File: functions.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * @return string, return string of html
+ */
+
+function addItemToHTML(array $dataFromQuery): string
+{
+    $result = '';
+    foreach ($dataFromQuery as $itemFromQuery) {
+        if ((!isset($itemFromQuery['variety'])) || (!isset($itemFromQuery['nameOfBrand'])) || (!isset($itemFromQuery['cost'])) || (!isset($itemFromQuery['country'])) || (!isset($itemFromQuery['tones']))) {
+            $result .= 'Error can\'t find key';
```
Comment (mporam) [inline functions.php]: if you pass in an array where the first sub array is valid, the second one is not and the third one is valid, you will get a very odd result from this... try it

File: index.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+<body>
+<div class="titleBar">
+    <h1>Wine Collection</h1>
+    <p>Let's make some pour decisions!</p>
+</div>
+
+<div class="container">
+    <?php
+    $wines=connect_to_db_and_extract_items('wineCollection');
+//    var_dump($wines);
```
Comment (mporam) [inline index.php]: remove debug code before submitting for review

File: index.php [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+</head>
+<body>
+<div class="titleBar">
+    <h1>Wine Collection</h1>
+    <p>Let's make some pour decisions!</p>
+</div>
+
+<div class="container">
+    <?php
+    $wines=connect_to_db_and_extract_items('wineCollection');
```
Comment (mporam) [inline index.php]: spaces either side of `=`

**Review by mporam — CHANGES_REQUESTED** [review]

File: functions.php (hunk 4 above)
Comment (ellie-r) [reply functions.php]: it prints the error to the screen inbetween the valid sub array, so I think I should break the loop if there is an error?

Comment (ellie-r) [reply functions.php]: i changed my mind, and have outputted from the function an array that contains the result of html and the error so they're not concatenated

**Review by mporam — CHANGES_REQUESTED** [review]
discussed changing addItemToHTML to return empty string if array is invalid

File: functions.php [hunk 7]
```diff
@@ -31,21 +31,17 @@ function extract_from_db(PDO $db): array {
...
 {
     $result = '';
-    $error = '';
     foreach ($dataFromQuery as $itemFromQuery) {
-        if ((!isset($itemFromQuery['variety'])) || (!isset($itemFromQuery['nameOfBrand']))
-            || (!isset($itemFromQuery['cost'])) || (!isset($itemFromQuery['country']))
-            || (!isset($itemFromQuery['tones']))) {
-            $error = 'Error can\'t find key';
-        } else {
+        if ((isset($itemFromQuery['variety'])) && (isset($itemFromQuery['nameOfBrand']))
```
Comment (mporam) [inline functions.php]: PSR says long if statements should look like:
```
if (
condition &&
condition &&
condition
)
```

File: functions.php [hunk 8]
```diff
@@ -31,21 +31,17 @@ function extract_from_db(PDO $db): array {
...
+            && (isset($itemFromQuery['cost'])) && (isset($itemFromQuery['country']))
+            && (isset($itemFromQuery['tones']))) {
             if (!isset($itemFromQuery['img_location'])) {
```
Comment (mporam) [inline functions.php:49]: have you tested if this works??

Comment (ellie-r) [reply functions.php:49]: yes, manually checked it with changing the db and theres a unit test that checks it

**Review by mporam — APPROVED** [review]

**Pull request merged**